rdrf/rdrf/forms/widgets/widgets.py

```diff
@@ -48,13 +48,21 @@ class OtherPleaseSpecifyWidget(MultiWidget):
     def usable_for_types():
         return {CDEDataTypes.STRING}
 
-    def __init__(self, main_choices, other_please_specify_value, unset_value, attrs=None):
+    def __init__(self, main_choices, other_please_specify_value, unset_value, attrs=None, widget_name=None):
         self.main_choices = main_choices
         self.other_please_specify_value = other_please_specify_value
         self.unset_value = unset_value
+        self.use_radio = widget_name == 'RadioSelect'
 
+        def default_main_widget():
+            return widgets.Select(attrs=attrs, choices=self.main_choices)
+
+        _main_widget_mapping = {
+            'RadioSelect': lambda: RadioSelect(attrs=attrs, choices=self.main_choices),
```

Comment:
I would map just to the widget class here to get rid of the lambda.
Same for `default_main_widget`. ie. `main_widget_cls = _main_widget_mapping.get(widget_name, widgets.Select`
and then you just construct it with:
`main_widget_cls(attrs=attrs, choices=self.main_choices)`.